Aztec/Classes/Extensions/String+RangeConversion.swift

```diff
@@ -241,4 +249,12 @@ public extension String {
 
         return startIndex ..< endIndex
     }
+
+    func isValidRange(_ range: Range<String.UTF16View.Index>) -> Bool {
+        isValidIndex(range.lowerBound) && isValidIndex(range.upperBound)
+    }
+
+    func isValidIndex(_ index: String.UTF16View.Index) -> Bool {
+        (self.startIndex ... self.endIndex).contains(index)
+    }
```

Comment:
[The `endIndex` is not an access index.](https://developer.apple.com/documentation/swift/string/endindex)

> A string’s “past the end” position—that is, the position one greater than the last valid subscript argument.